core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
+        } else if (!fir.isParam && sym.isJava) {
+          Nil
         } else {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
         }
       })
+      if (cir.fields.nonEmpty && putFields.isEmpty) {
+        throw PicklingException("No fields are captured. You need a custom pickler to handle this.")
```

Comment:
I think here, because you're targeting 0.10.x, you should mention how if this code previously compiled, then the genreated pickler was inaccurate and should not be trusted.